src/component/Uploader/Uploader.tsx

```diff
@@ -69,10 +75,16 @@ const Uploader = () => {
         enqueueSnackbar(msg, { variant: type });
       },
       onDropOver: (_e) => {
+        if (pathRef.current === defaultTrashPath) {
```

Comment:
Maybe we can improve this furthermore: use https://github.com/cloudreve/frontend/blob/master/src/component/FileManager/ContextMenu/useActionDisplayOpt.ts to determine if upload is enabled.
Something like (pseudo code):

```js
const parent = useAppSelector(
    (state) => state.fileManager[FileManagerIndex.main].list?.parent,
);
const uploadEnabled = useRef<Boolean>(false);
useEffect(()=>{
if (!parent) uploadEnabled.current = false;
const displayOpt = useActionDisplayOpt(
    [],
    ContextMenuTypes.empty,
    parent,
    FileManagerIndex.main,
 );
uploadEnabled.current=displayOpt.showUpload;
},[parent])
```

This will also disable DnD upload for not supported folders (e.g. folders without upload permission)